### PR TITLE
Unify daal4py and sklearnex building

### PR DESCRIPTION
**Pull request by Alexsandruss**
### Description

Merge daal4py building into sklearnex.

Changes:
- Merge residual of `setup_sklearnex.py` into `setup.py`, reassign `setup.py` from `daal4py` to `sklearnex` package
- Entirely remove `setup_sklearnex.py` commands from repo
- Update sklearnex `INSTALL.md` and delete daal4py one

---
- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.
- [x] I have resolved any merge conflicts that might occur with the base branch.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [x] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

**Review by icfaust — COMMENTED** [review]
Generally only a small recommendation. I'm surprised how little work this was (i.e. great work).  I will look closely at the INSTALL.md and README.md after the next changes for clarity since reading it from a changelog is difficult.  Ideally a follow up task would be for us to move away from setup.py to using the pyproject.toml, since setuptools setup.py installations have been deprecated.  If a ticket hasn't been made for this task, I would confer with me and @ethanglaser , since this would be more involved (not to be assigned to use, but have in the backlog).

File: .ci/pipeline/build-and-test-win.yml [hunk 1]
```diff
@@ -34,11 +34,10 @@ steps:
       call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvarsall" x64
       call activate CB
       set PREFIX=%CONDA_PREFIX%
+      set DALROOT=%CONDA_PREFIX%
```
[the remark this answers is not in the document]
Comment (Alexsandruss) [reply .ci/pipeline/build-and-test-win.yml]: Deleted.

**Review by ethanglaser — APPROVED** [review]
